cranelift/codegen/src/machinst/abi.rs

```diff
@@ -1495,6 +1507,10 @@ impl<M: ABIMachineSpec> Callee<M> {
                         &ABIArgSlot::Reg {
                             reg, ty, extension, ..
                         } => {
+                            reg_pairs.push(RetPair {
+                                vreg: from_reg,
+                                preg: Reg::from(reg),
+                            });
                             let from_bits = ty_bits(ty) as u8;
                             let ext = M::get_ext_mode(sigs[self.sig].call_conv, extension);
                             let reg: Writable<Reg> = Writable::from_reg(Reg::from(reg));
```

Comment:
This is going to make things a little interesting too: properly speaking, an extend should allocate a new temp for fully SSA output, rather than coerce the preg to a Writable and extend into it.

Reply:
@uweigand: I think that Chris's comment here is the same problem you noticed with extension on returns.

Reply:
This should be fixed now.